app/models/user.rb

```diff
@@ -0,0 +1,12 @@
+class User < ApplicationRecord
+  EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i
+
+  validates :username, presence: true, uniqueness: true, length: {minimum: 6}
+
+  validates :email, presence: true, uniqueness: true, format: EMAIL_REGEX
```

Comment:
any concern for case-insensitive uniqueness? example: do you want "GIJoe" and "gijoe" to be 2 different user accounts? what about "joe@example.test" and "Joe@Example.test"?